### PR TITLE
Fix: Include string literal keys in object-shorthand (Fixes #2374)

### LINKED ISSUE #2374: object-shorthand rule fails with quoted properties
.eslintrc:
```yaml
---
rules:
  object-shorthand: [2, always]
  no-unused-vars: 0
```

index.js:
```js
var o = { "a": "a" };
```

```bash
index.js
  1:10  error  Expected property shorthand  object-shorthand

✖ 1 problem (1 error, 0 warnings)
```

Shouldn't quoted properties only concern the `quote-props` rule?

### PR DESCRIPTION
**Pull request by xjamundx**
Quick fix for the issue mentioned above. Sorry this was so quick. Let me know if you'd like to see more tests. (Heading on :airplane:  in the morning)

### CODE REVIEW

**Comment by xjamundx** [comment]
Will add some "valid" cases with quotes to verify ;-)

File: lib/rules/object-shorthand.js [hunk 1]
```diff
@@ -56,6 +56,10 @@ module.exports = function(context) {
 
                 // {x: x} should be written as {x}
                 context.report(node, "Expected property shorthand.");
+            } else if (node.key.type === 'Literal' && node.key.value === node.value.name && APPLY_TO_PROPS) {
```
Comment (nzakas) [inline lib/rules/object-shorthand.js]: Double quotes :)

**Comment by xjamundx** [comment]
Yes, testing is helpful.....`"var x = {'a': 'a'}` fails...fixing.

File: lib/rules/object-shorthand.js (hunk 1 above)
Comment (xjamundx) [inline lib/rules/object-shorthand.js]: if only we had a linter .. .hahah sorry :runner: too fast

**Comment by xjamundx** [comment]
Found a few more edge cases, added tests, and of course ran the full test suite :)

**Comment by xjamundx** [comment]
2-3 more tests....

**Comment by AlexKVal** [comment]
I'm trying you patch on this file [react-bootstrap/src/styleMaps.js#L3](https://github.com/react-bootstrap/react-bootstrap/blob/master/src/styleMaps.js#L3)
On that line it warns me as it must.

Then I'm trying to refactor it into:
```js
CLASSES: {
    'alert', // <==  comma is at 3:11
```
and I've got this error
```console
$ eslint src/styleMaps.js

src/styleMaps.js
  3:11  error  Unexpected token
```

**Comment by AlexKVal** [comment]
I am sorry! 
I hadn't used eslint with exactly this last patch.

With it there are no warnings on that line (and such) at all and I had tried to do wrong refactoring.

**Comment by xjamundx** [comment]
I think this is ready to go

**Comment by xjamundx** [comment]
Maybe want to add something like `{["x"]: x}` but might be too much an edge case?

**Comment by AlexKVal** [comment]
If someone needs it, they can write about it with some usage cases to justify :wink: 
This is already awaited fix :+1:  :smile:

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**